pkgs/development/python-modules/jaxlib/bin.nix

```diff
@@ -146,15 +162,18 @@ buildPythonPackage {
   inherit version;
   format = "wheel";
 
-  disabled = !(pythonVersion == "3.9" || pythonVersion == "3.10" || pythonVersion == "3.11" || pythonVersion == "3.12");
+  # Note that the prebuilt jaxlib binary requires specific version of CUDA to
+  # work. The cuda12 jaxlib binaries only works with CUDA 12.2, and cuda11
+  # jaxlib binaries only works with CUDA 11.8.
+  disabled = !((cudaVersion == "11.8" && (pythonVersion == "3.9" || pythonVersion == "3.10" || pythonVersion == "3.11")) || (cudaVersion == "12.2" && (pythonVersion == "3.9" || pythonVersion == "3.10" || pythonVersion == "3.11" || pythonVersion == "3.12")));
```

Comment:
We should prob use `meta.broken` here instead of `disabled`. No idea why, but I've only ever seen `disabled` used for python version constraints and `meta.broken` used for other package constraints.

I also propose that we deduplicate the logic here: How about something like
```
gpuSrcs = {
  "cuda11.8-py3.9" = ...
  "cuda11.8-py3.10" = ...
  ...
};
gpuVersionString = "cuda{cudaVersion}-py{pythonVersion}";

meta.broken = !(gpuSrcs ? gpuVersionString);
```
? That way we don't duplicate the logic of what (cuda, python) versions are supported across both `gpuSrcs` and the boolean expression. Some care may be required to handle the case when `cudaPackagesGoogle = null`, and so forth.

Reply:
Good point. I have updated the logic here following the suggestion. Now `broken` is used to report violation of package constraints.

One small thing is that I was not able to write `(gpuSrcs ? gpuSrcVersionString)` in the expression for `broken`, because it will evaluate to `false`. However, explicitly writing `cuda${cudaVersion}-${pythonVersion}` there seems to work.

Reply:
Yes, I think `disabled = true` results in an error message saying something specifically about "this release of python not supporting XXX"

Reply:
I am currently using the `broken` approach, should I change it back?

Reply:
iiuc we're all in agreement on this